File: ClearFiveStages-HTML5/HelpScene.js
var HelpScene = function () {};

var PAGE_SPRITE_TAG	= 8000;
var TOTAL_PAGE_COUNT = 15;

HelpScene.prototype.onDidLoadFromCCB = function () {
    cc.SpriteFrameCache.getInstance().addSpriteFrames("tutorial.plist", "tutorial.png");
	this.page = 1;
    this.showHelpInfo();
    this.updatePageInfo();
};

HelpScene.prototype.onLeftArrowPressed = function () {
    if (this.page == 1) return;
    this.page --;
    this.showHelpInfo();
    this.updatePageInfo();
};

HelpScene.prototype.onRightArrowPressed = function () {
	if (this.page == TOTAL_PAGE_COUNT) return;
    this.page ++;
    this.showHelpInfo();
    this.updatePageInfo();
};

HelpScene.prototype.onHomePressed = function () {
    cc.Director.getInstance().popScene();
};

HelpScene.prototype.updatePageInfo = function () {
	this.currentPageLabel.setString(this.page);
	this.totalPageLabel.setString(TOTAL_PAGE_COUNT);
}

HelpScene.prototype.showHelpInfo = function () {
    var helpInfoSprite = this.rootNode.getChildByTag(999);
    this.rootNode.removeChildByTag(PAGE_SPRITE_TAG, true);
    this.titleLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_TUTORIAL_TITLE_PAGE_0 + this.page));
    this.subTitleLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_TUTORIAL_SUBTITLE_PAGE_0 + this.page));
    var ss = this.infoLabel.getDimensions();
    this.infoLabel.setDimensions(new cc.Size(300, 300));
    this.infoLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_TUTORIAL_INFO_PAGE_0 + this.page));
    // extra png
    if (this.page > 1 && this.page < 14) {
    	var winSize = cc.Director.getInstance().getWinSize();
		var pageSprite = new cc.Sprite();
		if (this.page == 2 || (this.page > 5 && this.page < 14)) {
            pageSprite.initWithSpriteFrameName(MultiLanguageUtil.getI18NResourceNameFrom("page{0}.png".format(this.page)));
		}
		if (this.page == 3 || this.page == 4 || this.page == 5) {
            pageSprite.initWithSpriteFrameName("page{0}.png".format(this.page));
        }
		pageSprite.setPosition(cc.p(winSize.width / 2, winSize.height / 3));
		this.rootNode.addChild(pageSprite, 999, PAGE_SPRITE_TAG);	
    }
};

File: ClearFiveStages-iOS/ClearFiveStages/Scene/HelpScene.m
#import "HelpScene.h"
#import "MultiLanguageUtil.h"
#import "SimpleAudioEngine.h"

#define PAGE_SPRITE_TAG	8000
#define TOTAL_PAGE_COUNT 15

@implementation HelpScene
@synthesize page;

-(void) didLoadFromCCB {
    [[CCSpriteFrameCache sharedSpriteFrameCache] addSpriteFramesWithFile:@"tutorial.plist"];
    self.page = 1;
    [self updatePageInfo];
    [self showHelpInfo];
}

-(void) onLeftArrowPressed {
	if (self.page == 1) return;
	self.page --;
    [self updatePageInfo];
	[self showHelpInfo];
}

-(void) onRightArrowPressed {
	if (self.page == 15) return;
	self.page ++;
    [self updatePageInfo];
	[self showHelpInfo];
}

-(void) onHomePressed {
    [[CCDirector sharedDirector] popScene];
}

-(void) updatePageInfo {
	[currentPageLabel setString:[NSString stringWithFormat:@"%d", self.page]];
	[totalPageLabel setString:[NSString stringWithFormat:@"%d", TOTAL_PAGE_COUNT]];
}

-(void) showHelpInfo {
	[self removeChildByTag:PAGE_SPRITE_TAG cleanup:YES];
    [titleLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_TUTORIAL_TITLE_PAGE_0 + self.page]];
    [subTitleLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_TUTORIAL_SUBTITLE_PAGE_0 + self.page]];
    [infoLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_TUTORIAL_INFO_PAGE_0 + self.page]];
    // extra png
    if (self.page > 1 && self.page < 14) {
    	CGSize winSize = [CCDirector sharedDirector].winSize;
		CCSprite *pageSprite = nil;
		if (self.page == 2 || (self.page > 5 && self.page < 14)) {
			pageSprite = [CCSprite spriteWithSpriteFrameName:[MultiLanguageUtil getI18NResourceNameFrom:[NSString stringWithFormat:@"page%d.png", self.page]]];
		}
		if (self.page == 3 || self.page == 4 || self.page == 5) {
			pageSprite = [CCSprite spriteWithSpriteFrameName:[NSString stringWithFormat:@"page%d.png", self.page]];
		}
        if (pageSprite) {
            pageSprite.position = ccp(winSize.width / 2, winSize.height / 3);
            [self addChild:pageSprite z:999 tag:PAGE_SPRITE_TAG];
        }
    }
}

@end
